Client/Frontend/Settings/AdvanceAccountSettingViewController.swift

```diff
@@ -0,0 +1,158 @@
+    }
+    
+    func setCustomSyncPrefs(_ data: Data, url: URL) {
+        
+        guard let json = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) as! [String:Any] else {
+            return
+        }
+        
+        self.profile.prefs.setBool(true, forKey: PrefsKeys.KeyUseCustomSyncService)
+        self.profile.prefs.setString(json["sync_tokenserver_base_url"] as! String, forKey: PrefsKeys.KeyCustomSyncToken)
+        self.profile.prefs.setString(json["profile_server_base_url"] as! String, forKey: PrefsKeys.KeyCustomSyncProfile)
+        self.profile.prefs.setString(json["oauth_server_base_url"] as! String, forKey: PrefsKeys.KeyCustomSyncOauth)
+        self.profile.prefs.setString(json["auth_server_base_url"] as! String, forKey: PrefsKeys.KeyCustomSyncAuth)
+        self.profile.prefs.setString(url.absoluteString, forKey: PrefsKeys.KeyCustomSyncWeb)
+        
+        self.profile.removeAccount()
+        
+        let alertController = UIAlertController(title: "", message: Strings.SettingsAdvanceAccountUrlUpdatedAlertMessage, preferredStyle: .alert)
+        let defaultAction = UIAlertAction(title: Strings.SettingsAdvanceAccountUrlUpdatedAlertOk, style: .default, handler: nil)
+        alertController.addAction(defaultAction)
+        
+        self.present(alertController, animated: true)
+    }
+    
+    func setCustomAccountPrefs(_ url: URL?) {
+        guard let url = url else {
+            // If the user attempts to set a nil url, clear all the custom service perferences
+            // and use default FxA servers.
+            self.clearCustomSyncPrefs()
+            return
+        }
+        
+        // FxA stores its server configuation under a well-known path. This attempts to download the configuration
+        // and save it into the users preferences.
+        let syncConfigureString = url.absoluteString + "/.well-known/fxa-client-configuration"
+        let syncConfigureURL = URL(string: syncConfigureString)!
+        
+        URLSession.shared.dataTask(with:syncConfigureURL, completionHandler: {(data, response, error) in
+            guard let data = data, error == nil else {
+                // Something went wrong while downloading or parsing the configuration.
+                let alertController = UIAlertController(title: Strings.SettingsAdvanceAccountUrlErrorAlertTitle, message: Strings.SettingsAdvanceAccountUrlErrorAlertMessage, preferredStyle: .alert)
+                let defaultAction = UIAlertAction(title: Strings.SettingsAdvanceAccountUrlErrorAlertOk, style: .default, handler: nil)
+                alertController.addAction(defaultAction)
+                
+                self.present(alertController, animated: true)
+                
+                return
+            }
+            self.setCustomSyncPrefs(data, url: url)
+        }).resume()
+    }
+    
+    override func generateSettings() -> [SettingSection] {
+        let prefs = profile.prefs
```

Comment:
You should probably be using a `BoolSetting` to toggle this on and off.

Changing the prefs only on *Set* is a bit weird from a UX. Have a look at how Home Page settings work.